src/generators.rs

```diff
@@ -181,6 +192,37 @@ impl BulletproofGens {
         }
     }
 
+    /// Increases the generators' capacity to the amount specified.
+    /// If less than or equal to the current capacity, does nothing.
+    pub fn increase_capacity(&mut self, new_capacity: usize) {
+        use byteorder::{ByteOrder, LittleEndian};
+
+        if self.gens_capacity >= new_capacity {
+            return;
+        }
+
+        for i in 0..self.party_capacity {
+            let party_index = i as u32;
+            let mut label = [b'G', 0, 0, 0, 0];
+            LittleEndian::write_u32(&mut label[1..5], party_index);
+            self.G_vec[i].append(
+                &mut GeneratorsChain::new(&label)
+                    .fast_forward(self.gens_capacity)
+                    .take(new_capacity - self.gens_capacity)
+                    .collect::<Vec<_>>(),
```

Comment:
if you use `.extend` instead of `.append`, you can use the iterator and avoid allocation caused by `.collect`:

https://doc.rust-lang.org/std/iter/trait.Extend.html

Reply:
```rust
            self.G_vec[i].extend(
	                GeneratorsChain::new(&label)
	                    .fast_forward(self.gens_capacity)
	                    .take(new_capacity - self.gens_capacity)
            );
```